hack/cherry_pick_pull.sh

```diff
@@ -0,0 +1,109 @@
+#!/bin/bash
+
+# Copyright 2015 The Kubernetes Authors All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Checkout a PR from GitHub. (Yes, this is sitting in a Git tree. How
+# meta.) Assumes you care about pulls from remote "upstream" and
+# checks thems out to a branch named pull_12345.
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+declare -r KUBE_ROOT="$(dirname "${BASH_SOURCE}")/.."
+cd "${KUBE_ROOT}"
+
+declare -r STARTINGBRANCH=$(git symbolic-ref --short HEAD)
+
+if [[ "$#" -ne 2 ]]; then
+  echo "${0} <pr-number> <branch>: cherry pick <pr> onto <branch> and leave instructions for proposing pull request"
+  exit 2
+fi
+
+if git_status=$(git status --porcelain 2>/dev/null) && [[ -n ${git_status} ]]; then
+  echo "!!! Dirty tree. Clean up and try again."
+  exit 1
+fi
+
+declare -r PULL="${1}"
+declare -r BRANCH="${2}"
+echo "+++ Updating remotes..."
+git remote update
+
+echo "+++ Downloading patch to /tmp/${PULL}.patch (in case you need to do this again)"
+
+curl -o "/tmp/${PULL}.patch" -sSL "https://github.com/GoogleCloudPlatform/kubernetes/pull/${PULL}.patch"
+
+declare -r NEWBRANCH="$(echo automated-cherry-pick-of-#${PULL}-on-${BRANCH} | sed 's/\//-/g')"
+declare -r NEWBRANCHUNIQ="${NEWBRANCH}-$(date +%s)"
+echo "+++ Creating local branch ${NEWBRANCHUNIQ}"
+
+function return_to_kansas {
+  git checkout -f "${STARTINGBRANCH}"
+}
+trap return_to_kansas EXIT
+
+git checkout -b "${NEWBRANCHUNIQ}" "${BRANCH}"
+
+echo
+echo "+++ About to attempt cherry pick of PR. To reattempt:"
+echo "  $ git am -3 /tmp/${PULL}.patch"
+echo
+git am -3 "/tmp/${PULL}.patch" || {
+  while unmerged=$(git status --porcelain | grep ^U) && [[ -n ${unmerged} ]]; do
+    echo
+    echo "+++ Conflicts detected:"
+    echo
+    git status --porcelain | grep ^U
+    echo
+    echo "+++ Please resolve the conflicts in another window (and remember to 'git add')"
+    read -p "+++ Proceed (anything but 'y' aborts the cherry-pick)? [y/n] " -r
+    echo
+    [[ $REPLY =~ ^[yY]$ ]] || {
```

Comment:
Heh. Hoisted by my own `${NIT}` because of cut & paste from some other confirmation script. :)